server/test/durablejobs/DurableJobSchedulerTest.java

```diff
@@ -0,0 +1,71 @@
+package durablejobs;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Clock;
+import java.time.Instant;
+import models.PersistedDurableJob;
+import org.junit.Before;
+import org.junit.Test;
+import repository.PersistedDurableJobRepository;
+import repository.ResetPostgres;
+
+public class DurableJobSchedulerTest extends ResetPostgres {
+
+  private Clock clock;
+  private DurableJobRegistry durableJobRegistry;
+  private PersistedDurableJobRepository persistedDurableJobRepository;
+  private RecurringJobScheduler recurringJobScheduler;
+
+  @Before
+  public void setUp() {
+    clock = instanceOf(Clock.class);
+    durableJobRegistry = new DurableJobRegistry();
+    durableJobRegistry.register(
+        DurableJobName.TEST,
+        (persistedDurableJob) -> makeTestJob(persistedDurableJob, () -> {}),
+        new RecurringJobExecutionTimeResolvers.Sunday2Am());
+
+    persistedDurableJobRepository = instanceOf(PersistedDurableJobRepository.class);
+    recurringJobScheduler =
+        new RecurringJobScheduler(clock, durableJobRegistry, persistedDurableJobRepository);
+  }
+
+  @Test
+  public void scheduleRecurringJobs_schedulesUnscheduledRecurringJobs() {
+    var jobs = persistedDurableJobRepository.getJobs();
+    assertThat(jobs.size()).isZero();
+
+    recurringJobScheduler.scheduleRecurringJobs();
+
+    jobs = persistedDurableJobRepository.getJobs();
+    assertThat(jobs.size()).isEqualTo(1);
+    var job = jobs.get(0);
+    Instant expectedExecutionTime =
+        new RecurringJobExecutionTimeResolvers.Sunday2Am().resolveExecutionTime(clock);
+    assertThat(job.getExecutionTime()).isEqualTo(expectedExecutionTime);
+    assertThat(job.getJobName()).isEqualTo(DurableJobName.TEST.getJobNameString());
+
+    // Assert that subsequent runs do not create duplicates
+    recurringJobScheduler.scheduleRecurringJobs();
+    jobs = persistedDurableJobRepository.getJobs();
+    assertThat(jobs.size()).isEqualTo(1);
+    assertThat(job.getExecutionTime()).isEqualTo(expectedExecutionTime);
+    assertThat(job.getJobName()).isEqualTo(DurableJobName.TEST.getJobNameString());
+  }
+
```

Comment:
note to reviewer: i made an effort at testing the retry logic and ran into an issue with mocking the transaction object -- it's an interface, so mockito can't call through to the real implementation, making it much more difficult to test.

here's the attempt, other ideas on how to test the retry logic are welcome otherwise we'll go without coverage

```java
  @Test
  public void scheduleRecurringJobs_retriesFailedInsertions() {
    try (MockedStatic<DB> mockDb = mockStatic(DB.class)) {
      var mockTransactionA = mock(Transaction.class);
      var mockDatabase = mock(Database.class);
      mockDb.when(DB::getDefault).thenReturn(mockDatabase);
      when(mockDatabase.beginTransaction())
          .thenReturn(mockTransactionA)
          .thenAnswer((Answer<Transaction>) invocation -> DB.beginTransaction());

      doThrow(OptimisticLockException.class).when(mockTransactionA).commit();

      var jobs = persistedDurableJobRepository.getJobs();
      assertThat(jobs.size()).isZero();

      recurringJobScheduler.scheduleRecurringJobs();

      jobs = persistedDurableJobRepository.getJobs();
      assertThat(jobs.size()).isEqualTo(1);
      var job = jobs.get(0);
      Instant expectedExecutionTime =
          new RecurringJobExecutionTimeResolvers.Sunday2Am().resolveExecutionTime(clock);
      assertThat(job.getExecutionTime()).isEqualTo(expectedExecutionTime);
      assertThat(job.getJobName()).isEqualTo(DurableJobName.TEST.getJobNameString());
    }
  }
```